isvcs/manager.go

```diff
@@ -157,9 +159,20 @@ func (m *Manager) GetHealthStatus(name string) (IServiceHealthResult, error) {
 	defer svc.lock.RUnlock()
 
 	result.ContainerName = svc.name()
-	for _, value := range svc.healthStatuses {
-		result.HealthStatuses = append(result.HealthStatuses, *value)
+
+	if instIndex == HEALTH_STATUS_INDEX_ALL {
+		for _, statusMap := range svc.healthStatuses {
+			for _, value := range statusMap {
+				result.HealthStatuses = append(result.HealthStatuses, *value)
+			}
+		}
+	} else {
+		for _, value := range svc.healthStatuses[instIndex] {
```

Comment:
Before using `instIndex`, there should be some kind of validation check to make sure that `instIndex` is in the valid range of `svc.healthStatuses` (i.e. not too big)